### PR TITLE
add josh instructions

### PR DESCRIPTION
**Pull request by RalfJung**
This explains how to sync Miri changes to and from rustc.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -275,3 +275,52 @@ see <https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html>.
...
 With this, you should now have a working development setup! See
 [above](#building-and-testing-miri) for how to proceed working on Miri.
+
+## Advanced topic: Syncing with the rustc repo
+
+We use the [`josh` proxy](https://github.com/josh-project/josh) to transmit
+changes between the rustc and Miri repositories. For now, josh needs to be built
+from source. This downloads and runs josh:
+
+```rust
```
Comment (saethlin) [inline CONTRIBUTING.md:285]: Is this Rust code?

**Comment by saethlin** [comment]
I've read over this and it makes sense, though of course I've never tried to actually use josh.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -275,3 +275,52 @@ see <https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html>.
...
+git clone https://github.com/josh-project/josh
+cd josh
+git checkout @changes/master/christian.schilling.de@gmail.com/start-filter
```
Comment (oli-obk) [inline CONTRIBUTING.md:288]: What is that filter 😃

Comment (oli-obk) [reply CONTRIBUTING.md:288]: Oh that's a branch

File: CONTRIBUTING.md (hunk 1 above)
Comment (RalfJung) [reply CONTRIBUTING.md:285]: I wish it was. :joy:

File: CONTRIBUTING.md (hunk 2 above)
Comment (RalfJung) [reply CONTRIBUTING.md:288]: Yeah, it's the josh feature branch for the `at_commit` filter.

That filter is also a bit weird, but this is not it. ;)

**Pull request merged**